packages/chat-ui/src/hook/use-workflow/index.tsx

```diff
@@ -0,0 +1,240 @@
+import { useState, useEffect, useCallback, useMemo, useRef } from 'react'
+import { WorkflowSDK } from './sdk'
+
+export interface WorkflowEvent {
+  name: string
+  [key: string]: any
+}
+
+export type TaskStatus = 'idle' | 'running' | 'complete' | 'error'
+
+export interface WorkflowHookParams<O extends WorkflowEvent = WorkflowEvent> {
+  baseUrl?: string // Optional base URL for the workflow API
+  workflow: string // Name of the registered deployment
+  sessionId?: string // Optional session ID for resuming a workflow session
+  taskId?: string // Optional task ID for resuming a workflow task
+  initialTaskCallbacks?: TaskCallbacks<O> // Optional callbacks for the initial task if taskId is provided
+}
+
+export interface TaskCallbacks<O extends WorkflowEvent = WorkflowEvent> {
+  onStop?: (events: O[]) => void
```

Comment:
still not sure whether we have multiple events here? @leehuwuj ?
```suggestion
  onStopEvent?: (events: O) => void
```